service/storage_service.cc

```diff
@@ -3872,6 +3872,9 @@ future<> storage_service::on_remove(gms::inet_address endpoint, gms::permit_id p
     auto tmlock = co_await get_token_metadata_lock();
     auto tmptr = co_await get_mutable_token_metadata_ptr();
     tmptr->remove_endpoint(endpoint);
+    if (const auto host_id = tmptr->get_new()->get_host_id_if_known(endpoint); host_id) {
```

Comment:
What are the scenarios in which host_id could not be found?
Or could it happen that we find it, but it's not the right one?